static/js/common/upload-addon.js

```diff
@@ -429,7 +429,8 @@
                     $("<strong>").text(message).appendTo(upload_results);
 
                     // Specific messages for unlisted addons.
-                    if (isUnlisted()) {
+                    var validation_type = results.validation.detected_type
+                    if ((validation_type == 'extension' || validation_type == 'dictionary' || validation_type == 'languagepack') && isUnlisted()) {
```

Comment:
You can do `if (["extension", "dictionary", "languagepack"].includes(validation_type) && isUnlisted())` (I think 😆 - my javascript is really rusty)
We generally use double quotes `"` in our javascript, by convention.